src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -210,9 +210,13 @@ public boolean publish(String message, JSONArray attachments, String color) {
                 HttpEntity entity = response.getEntity();
                 if (botUser && entity != null) {
                     responseString = EntityUtils.toString(entity);
+                    try {
 
-                    org.json.JSONObject slackResponse = new org.json.JSONObject(responseString);
-                    result = slackResponse.getBoolean("ok");
+                        org.json.JSONObject slackResponse = new org.json.JSONObject(responseString);
+                        result = slackResponse.getBoolean("ok");
+                    } catch (org.json.JSONException ex) {
+                        result = false;
```

Comment:
can you at least log the response please? perhaps adding an entry to the build log and if fail build is true then rethrowing?

I would prefer dodgy config is caught higher but this might patch it a little bit

Reply:
OK I added a log for the exception.

Reply:
There is an additional log here: https://github.com/jenkinsci/slack-plugin/pull/598/files#diff-282db0eb13a24d1729df3e194a35f6d7R223

if `result` is false.  So at least this should remove large stacktraces, but at least give more useful diagnostics in the logs as to what is going wrong and how to fix it.